commit 1bf8012fc6997f2117f6919369cde16659db60e0
Author: Wen Yang <wen.yang@linux.dev>
Date:   Mon Aug 19 22:59:45 2024 +0800

    pstore: replace spinlock_t by raw_spinlock_t
    
    pstore_dump() is called when both preemption and local IRQ are disabled,
    and a spinlock is obtained, which is problematic for the RT kernel because
    in this configuration, spinlocks are sleep locks.
    
    Replace the spinlock_t with raw_spinlock_t to avoid sleeping in atomic context.
    
    Signed-off-by: Wen Yang <wen.yang@linux.dev>
    Cc: Kees Cook <kees@kernel.org>
    Cc: Tony Luck <tony.luck@intel.com>
    Cc: Guilherme G. Piccoli <gpiccoli@igalia.com>
    Cc: linux-hardening@vger.kernel.org
    Cc: linux-kernel@vger.kernel.org
    Link: https://lore.kernel.org/r/20240819145945.61274-1-wen.yang@linux.dev
    Signed-off-by: Kees Cook <kees@kernel.org>

diff --git a/include/linux/pstore.h b/include/linux/pstore.h
index 638507a3c8ff..fed601053c51 100644
--- a/include/linux/pstore.h
+++ b/include/linux/pstore.h
@@ -182,7 +182,7 @@ struct pstore_info {
 	struct module	*owner;
 	const char	*name;
 
-	spinlock_t	buf_lock;
+	raw_spinlock_t	buf_lock;
 	char		*buf;
 	size_t		bufsize;